### ISSUE TITLE
Local notification plug-in on iOS11

### ISSUE THREAD

**Issue by KristjanHjartarson** [sabrinathai/ExperienceSampler#21]
The local notification plug-in by Katzer does not work on the newly released iOS11.

The app runs smoothly but the participant does not receive any notifications or only sporadically at best. 

Consequently, we are not able to run the app on the Iphone in our ongoing study anymore. 

Do you know if Katze intends to update the plug-in? If not, are there any viable alternative plug ins that you know of?

Sincerely, 
Kristjan

**Comment by sabrinathai** [comment]
Hi Kristjan, 

So it looks like there is an updated version of the katzer plugin that is compatible with iOS11, but it's still under development and the documentation is not up to date yet. You can find some information about it here (https://github.com/katzer/cordova-plugin-local-notifications/issues/1370). 

Alternatively, someone posted that they have had some success using this plugin with iOS11 (https://github.com/DavidBriglio/cordova-ios10-simple-notification). 

Let me know which solution worked best for you, so I can update the ExperienceSampler documentation.

Cheers,
Sabrina

**Comment by KristjanHjartarson** [comment]
Hi Sabrina

Thank you for a very swift reply. 

We will look into the options that you mentioned. 

I will keep you updated on how it goes. 

Sincerely, 
Kristjan

**Comment by sabrinathai** [comment]
Hi Kristjan,
I recently tested the existing version of the katzer local notification in on my iPhone with iOS11, and it has been working fine. Have you had any luck with the other plugins or with getting the original one to work? I wonder why the original one worked for me but not for you.

Cheers,
Sabrina

**Comment by KristjanHjartarson** [comment]
Hi Sabrina

We have not been able to get the plugin to work nor the alternative plugin mentioned in the link you provided. 

Which version of the Katzer plugin are you referring to. The "x" branch or the master?

May I ask which version of xcode, swift and other relevant software you use when building the app?

Sincerely, 
Kristján

**Comment by sabrinathai** [comment]
Hi Kristján, 

I installed the 0.8.5 version of the katzer plugin. I'm using Xcode version 9.0, and cordova version 7.1.0. I hope that helps. 

If that doesn't work, you can send me email with a dropbox link to all the cordova files and I can take a look.

Cheers,
Sabrina

**Comment by KristjanHjartarson** [comment]
Hi Sabrina

We were able to solve the issue by applying the new triggers in the x-branch of the plug-in as well as defining the priority of the notification (priority=1). Both has since been added to the new 0.9 beta of the plug-in if i´m not mistaken. 

Thank you for all your help!

I do, however, have one more question. You recently changed the doGet function to doPost in the google collector script. Was that because of the limited storage capacity of the doGet function? We have had some instances where the app loses its connection to the google data server and is not able to send the data. We are currently using the doGet function. Might that be relevant? 

Sincerely,
Kristjan

**Comment by sabrinathai** [comment]
Hi Kristjan,

Thanks for letting me know about the solution! I'll try to update the documentation as soon as I can. 

Yes! That was one of the main reasons I updated from doGet to doPost. When we first developed the Google Script, for some reason it would not let us use the doPost method, but recently, I was able to use it. In the past, I have noticed some data loss because of using doGet and the Google Script. But when we used doPost with our server, we also lost some of our data usually with Android phones. I have recently collected data using the doPost with the Google script, so I can let you know if that has improved the data sending. But you also think it has improved, I'd love to hear from you too. 

The one thing I have noticed is that doPost does take a bit longer to post the data to Google. So I would just let participants know that they should be patient. 

Cheers,
Sabrina

**Comment by KristjanHjartarson** [comment]
Sabrina, thanks again for all the help. 

I would love to know if using the doPost method with the google script has improved the data sending for you. 

If we decide to change the method to doPost i´ll be sure to let you know how it works for us. 

Sincerely, 
Kristjan

**Comment by sabrinathai** [comment]
Hi Kristjan,

Sorry it took me a while to get back to you. I just restructured the new data I collected using the doPost data. We completed data from ~45 variables, and at most I only lost data from the snooze question, but that value is easy to generate. 

I think if you collect more than 45 variables at one time, I would set the app to save data half way through the survey and then save the rest of the data at the end of the survey. I hope that helps. 

In a second dataset, I collected data from 21 variables each time, and there was no data loss at all using the doPost method. 

Cheers,
Sabrina
